src/uct/ib/dc/accel/dc_mlx5.c

```diff
@@ -571,11 +576,37 @@ ucs_status_t uct_dc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
     UCT_DC_CHECK_RES(&iface->super, dc_ep);
     UCT_DC_MLX5_IFACE_TXQP_GET(iface, dc_ep, txqp, txwq);
 
+    dc_req = ucs_derived_of(req, uct_dc_fc_request_t);
+
     if (op == UCT_RC_EP_FC_PURE_GRANT) {
         ucs_assert(req != NULL);
-        dc_req    = ucs_derived_of(req, uct_dc_fc_request_t);
-        sender_ep = (uintptr_t)dc_req->sender_ep;
+
         ib_iface  = &iface->super.super.super;
+        sender_ep = (uintptr_t)dc_req->sender.ep;
+
+        /* TODO: loot at common code with uct_ud_mlx5_iface_get_av */
```

Comment:
yep, need to add a function to ib/mlx5 which takes ah_attr argument and fills a 'struct mlx5_wqe_av'

Reply:
ok, will refactor this in separate PR (it will affect several functions where ah used)